fanficfare/adapters/adapter_silmarillionwritersguildorg.py

```diff
@@ -0,0 +1,236 @@
+        # print data
+
+        # Now go hunting for all the meta data and the chapter list.
+
+        ## Title and author
+        a = soup.find('h6')
+
+        titlelinks = a.find_all('a')
+        aut= titlelinks[1]
+        
+        self.story.setMetadata('authorId',aut['href'].split('=')[1])
+        self.story.setMetadata('authorUrl','https://'+self.host+'/archive/home/'+aut['href'])
+        self.story.setMetadata('author',aut.string)
+        asoup = self.make_soup(self._fetchUrl(self.story.getMetadata('authorUrl')))
+
+        self.story.setMetadata('title',a.find('strong').find('a').get_text())
+        
+        try:
+            # Find Series name from series URL.
+            a = soup.find('a', href=re.compile(r"viewseries.php\?seriesid=\d+"))
+            series_name = a.string
+            series_url = 'https://'+self.host+'/archive/home/'+a['href']
+
+            # use BeautifulSoup HTML parser to make everything easier to find.
+            seriessoup = self_make_soup(self._fetchUrl(series_url))
+            storyas = seriessoup.findAll('a', href=re.compile(r'^viewstory.php\?sid=\d+$'))
+            i=1
+            for a in storyas:
+                if a['href'] == ('viewstory.php?sid='+self.story.getMetadata('storyId')):
+                    self.setSeries(series_name, i)
+                    self.story.setMetadata('seriesUrl',series_url)
+                    break
+                i+=1
+
+        except:
+            # I find it hard to care if the series parsing fails
+            pass
+
+        # Find the chapters by regexing urls
+        chapters=soup.findAll('a', href=re.compile(r'viewstory.php\?sid='+self.story.getMetadata('storyId')+"&chapter=\d+$"))
+        
+        #logger.debug(chapters)
+        
+        if len(chapters)==1:
+            self.add_chapter(self.story.getMetadata('title'),'https://'+self.host+'/archive/home/'+chapters[0]['href'])
+        else:
+            for chapter in chapters:
+                logger.debug("Added Chapter: "+chapter.string)
+                self.add_chapter(chapter,'https://'+self.host+'/archive/home/'+chapter['href'])
+                
+        
+
+	# find the details section for the work
+	
+        workDetails = soup.find('div', {'id' : 'general'}).find('div', {'id' : 'general'})
+        
+        # Find the summary 
+        summary = workDetails.find_all('p')[3]
+        summary.name='div' # change td to div.  Makes Calibre
+                           # sanitize_html() happier when description
+                           # is empty.
+        self.setDescription(url,summary)
+        
+        # no convenient way to extract metadata so bodge it by finding relevant identifier string and using next element as the data source
+
+        #get characters
+        charList = workDetails.findAll('a', href=re.compile(r'browse.php?type=characters'+"&charid=\d+$"))
+        charText = [char.string for char in charList]
+        self.chararacters = ', '.join(charText)
```

Comment:
I assume these `self.chararacters`(sic), etc are for debugging in your dev environment?  Because they aren't needed for FFF.  And you're calling `.string` on each tag twice this way.  You could call `self.story.extendList(charText)` instead of looping again.  (I'd also rename `charText`, etc since it's a list.)

Reply:
Oops my apologies, I must have forgotten to remove these after debugging! have now removed them and renamed charText to charList.